scripts/pipeline/copy_data.pl

```diff
@@ -709,16 +686,28 @@ sub copy_genomic_align_blocks {
   my @copy_data_args = (undef, 'skip_disable_keys');
 
   #copy genomic_align_block table
-  if ($fix_gab or $fix_gab_gid) {
+  if ($fix_gab) {
     copy_data($from_dbc, $to_dbc,
        "genomic_align_block",
-       "SELECT genomic_align_block_id+$fix_gab, method_link_species_set_id, score, perc_id, length, group_id+$fix_gab_gid, level_id, direction".
+       "SELECT genomic_align_block_id+$fix_gab, method_link_species_set_id, score, perc_id, length, group_id, level_id, direction" .
          " FROM genomic_align_block WHERE method_link_species_set_id = $mlss_id",
        @copy_data_args);
   } else {
     copy_table($from_dbc, $to_dbc, 'genomic_align_block', "method_link_species_set_id = $mlss_id", undef, "skip_disable_keys");
   }
 
+  # Fix the group_id when it is not in its MLSS id range
+  if ($fix_gab_gid) {
+      $to_dbc->do("UPDATE genomic_align_block
+          SET group_id = CAST(CONCAT(method_link_species_set_id, LPAD(group_id, 10, '0')) AS UNSIGNED INTEGER)
```

Comment:
Just want to check that this is the behaviour I am seeing: `LPAD` will pad the existing `group_id` to a length of 10 digits, and the `method_link_species_set_id` will essentially be prepended to this in the `CONCAT` step, if the `group_id` is less than `10000000000` - then as a form of overkill, the next update cuts it down to size/ups the size when not between the constraints?

Reply:
You got it right. As a note, I wouldn't call the second update an overkill since those IDs below 10^10 cannot be updated using `RIGHT` (Arthur did a great job noticing this and including it in his notes).

Reply:
I didn't spot that :+1: 
But in any case, sanity satisfied :smile: